### PR TITLE
Add dep to compile assets via webpack

### PR DESCRIPTION
**Pull request by nickbrowne**
I think it works!

### CODE REVIEW

**Comment by nickbrowne** [comment]
[Add dep to compile webpack assets](https://trello.com/c/q7CQWkWp/6646-add-dep-to-compile-webpack-assets)

File: deploy.rb [hunk 1]
```diff
@@ -6,6 +6,12 @@ def on_standby?
   current_hostname == standby_hostname
 end
 
+dep 'webpack compile during deploy', :env, :deploying, template: 'task' do
+  requires 'npm packages installed'
```
Comment (yob) [inline deploy.rb]: This dep lives in `sharejs.rb`. All the deps in this repo are in a global namespace so in most ways it doesn't really matter, but maybe it makes sense to move that dep into this file if it's going to be used across multiple projects?

Comment (nickbrowne) [inline deploy.rb]: hmm good point, I didn't even noice that!

File: deploy.rb [hunk 2]
```diff
@@ -6,6 +6,23 @@ def on_standby?
   current_hostname == standby_hostname
 end
 
+dep 'npm packages installed', :path do
+  met? {
+    output = raw_shell('npm ls', :cd => path)
+    # Older `npm` versions exit 0 on failure.
```
Comment (yob) [inline deploy.rb]: 🤦‍♂️

Comment (nickbrowne) [inline deploy.rb]: You only live to only live once once

**Comment by yob** [comment]
seems legit 👍

**Pull request merged**